### ISSUE TITLE
Clarify StatLearn docs

### ISSUE THREAD

**Issue by andreasnoack** [joshday/OnlineStats.jl#110]
It's not clear how the parameters and regressors/features enter the model. You specify a loss function but that just tells you about the mapping from predicted values, outcome/labels, and parameters to a loss but not how the prediction is constructed. I assume it is just the linear predictor but it would be good to clarify in the docs.

**Comment by joshday** [comment]
Here is the current docstring.  Still needs to get filled out a bit I think.

```
help?> StatLearn
search: StatLearn

  StatLearn(p, args...)

  Fit a model that is linear in the parameters.

  The (offline) objective function that StatLearn approximately minimizes is

  (1/n) ∑ᵢ f(yᵢ, xᵢ'β) + ∑ⱼ λⱼ g(βⱼ),

  where fᵢ are loss functions of a single response and linear predictor, λⱼs are nonnegative regularization parameters, and g is a penalty function.

     Arguments
    ≡≡≡≡≡≡≡≡≡≡≡

    •    loss = .5 * L2DistLoss()

    •    penalty = NoPenalty()

    •    algorithm = SGD()

    •    rate = LearningRate(.6) (keyword arg)


     Example
    ≡≡≡≡≡≡≡≡≡

  x = randn(1000, 5)
  y = x * linspace(-1, 1, 5) + randn(1000)

  o = fit!(StatLearn(5, MSPI()), (x, y))
  coef(o)
```
